pkg/opensearch/lucene_handler.go

```diff
@@ -284,6 +290,32 @@ func (h *luceneHandler) executeQueries(ctx context.Context) (*backend.QueryDataR
 	return rp.parseResponse()
 }
 
+// getParametersFromServiceMapResult extracts the lists of services and operations from the
+// response to the Prefetch request. These will be used to build the subsequent Stats request.
+func getParametersFromServiceMapResult(smResult *client.SearchResponse) ([]string, []string) {
+	services := make([]string, 0)
+	operationMap := make(map[string]bool)
+
+	buckets := smResult.Aggregations["service_name"].(map[string]interface{})["buckets"].([]interface{})
+	for _, bucket := range buckets {
+		service := bucket.(map[string]interface{})
+
+		services = append(services, service["key"].(string))
+		targets := service["target_domain"].(map[string]interface{})["buckets"].([]interface{})
+		for _, targetDomain := range targets {
+			targetResources := targetDomain.(map[string]interface{})["target_resource"].(map[string]interface{})["buckets"].([]interface{})
+			for _, targetResource := range targetResources {
+				operationMap[targetResource.(map[string]interface{})["key"].(string)] = true
```

Comment:
I don't know golang well enough to say this with full confidence but it feels like there must be a way somehow to do this that is not constantly casting stuff, or maybe this is just how golang is lol? Then again I feel like that could be fixed in a subsequent pr probs since this is all just in one little func.

Reply:
I have a test branch somewhere that provides helpers to do this in a nicer way, would love to get that in soon!